testsuite/forge/src/backend/k8s/cluster_helper.rs

```diff
@@ -61,84 +72,86 @@ pub fn set_validator_image_tag(
+        "-p",
+        "{\"metadata\":{\"labels\":{\"status\":\"deployed\"}}}",
     ];
-    let validator_upgrade_args = [&validator_upgrade_base_args, options].concat();
-    println!("{:?}", validator_upgrade_args);
-    let validator_upgrade_output = Command::new(HELM_BIN)
+    println!("{:?}", helm_patch_args);
+    let helm_patch_output = Command::new(KUBECTL_BIN)
         .stdout(Stdio::inherit())
-        .args(&validator_upgrade_args)
+        .args(&helm_patch_args)
         .output()
-        .expect("failed to helm upgrade valNN");
-    if !validator_upgrade_output.status.success() {
+        .expect("failed to kubectl patch secret valNN");
+    assert!(
+        helm_patch_output.status.success(),
+        "{}",
+        String::from_utf8(helm_patch_output.stderr).unwrap()
+    );
+
+    Ok(())
+}
+
+fn upgrade_helm_release(release_name: &str, helm_chart: &str, options: &[&str]) -> Result<()> {
+    let upgrade_base_args = ["upgrade", release_name, helm_chart];
+    let upgrade_args = [&upgrade_base_args, options].concat();
+    println!("{:?}", upgrade_args);
+    let upgrade_output = Command::new(HELM_BIN)
+        .stdout(Stdio::inherit())
+        .args(&upgrade_args)
+        .output()
+        .unwrap_or_else(|_| {
+            panic!(
+                "failed to helm upgrade release {} with chart {}",
+                release_name, helm_chart
+            )
+        });
+    if !upgrade_output.status.success() {
         bail!(format!(
             "Upgrade not completed: {}",
-            String::from_utf8(validator_upgrade_output.stderr).unwrap()
+            String::from_utf8(upgrade_output.stderr).unwrap()
         ));
     }
 
     Ok(())
 }
 
-fn upgrade_testnet(
-    helm_repo: &str,
-    era_num: &str,
-    num_validator: usize,
-    image_tag: &str,
-) -> Result<()> {
-    // upgrade testnet
-    let testnet_upgrade_args = [
-        "upgrade",
-        "diem",
-        &format!("{}/testnet", helm_repo),
-        "--reuse-values",
-        "--history-max",
-        "2",
-        "--set",
-        &format!("genesis.era={}", era_num),
-        "--set",
-        &format!("genesis.numValidators={}", num_validator),
-        "--set",
-        &format!("imageTag={}", image_tag),
-    ];
-    println!("{:?}", testnet_upgrade_args);
-    let testnet_upgrade_output = Command::new(HELM_BIN)
-        .stdout(Stdio::inherit())
-        .args(&testnet_upgrade_args)
-        .output()
-        .expect("failed to helm upgrade diem");
-    assert!(
-        testnet_upgrade_output.status.success(),
-        "{}",
-        String::from_utf8(testnet_upgrade_output.stderr).unwrap()
-    );
-    println!("Testnet upgraded");
+fn upgrade_validator(validator_name: &str, helm_repo: &str, options: &[&str]) -> Result<()> {
+    upgrade_helm_release(
+        validator_name,
+        &format!("{}/diem-validator", helm_repo),
+        options,
+    )
+}
 
-    Ok(())
+fn upgrade_testnet(helm_repo: &str, options: &[&str]) -> Result<()> {
```

Comment:
I think i'd like for the options to be a bit more strongly typed than just a slice of strings but given this is an internal to this file api it's fine as is.